examples/ticks_and_spines/date_concise_formatter.py

```diff
@@ -0,0 +1,214 @@
+    ax.plot(dates, y)
+    ax.set_xlim(lims[nn])
+    # rotate_labels...
+    for label in ax.get_xticklabels():
+        label.set_rotation(40)
+        label.set_horizontalalignment('right')
+axs[0].set_title('Default Date Formatter')
+plt.show()
+
+#############################################################################
+# The default date formater is quite verbose, so Matplotlib has an
+# alternative that can be chosen with a few extra lines of code.  Note,
+# for these axes the labels do not need to be rotated as they do for the
+# default formatter because the labels are as small as possible.
+
+fig, axs = plt.subplots(3, 1, constrained_layout=True, figsize=(6, 6))
+for nn, ax in enumerate(axs):
+    locator = mdates.AutoDateLocator(minticks=3, maxticks=7)
+    formatter = mdates.ConciseDateFormatter(locator)
+    ax.xaxis.set_major_locator(locator)
+    ax.xaxis.set_major_formatter(formatter)
+
+    ax.plot(dates, y)
+    ax.set_xlim(lims[nn])
+axs[0].set_title('Concise Date Formatter')
+
+plt.show()
+
+#############################################################################
+# If all calls to axes that have dates are to be made using this converter,
+# it is probably most convenient to use the units reigistry where you do
+# imports:
+
+import matplotlib.units as munits
+munits.registry[np.datetime64] = mdates.ConciseDateConverter()
+munits.registry[datetime.date] = mdates.ConciseDateConverter()
+munits.registry[datetime.datetime] = mdates.ConciseDateConverter()
+
+fig, axs = plt.subplots(3, 1, figsize=(6, 6), constrained_layout=True)
+for nn, ax in enumerate(axs):
+    ax.plot(dates, y)
+    ax.set_xlim(lims[nn])
+axs[0].set_title('Concise Date Formatter')
+
+plt.show()
+
+#############################################################################
+# Localization of date formats
+# ============================
+#
+# Dates formats can be localized if the default formats are not desirable by
+# manipulating one of three lists of strings.
+#
+# The ``formatter.formats`` list of formats is for the normal tick labels,
+# There are seven levels, years, months, days, hours, minutes, seconds, and
+# microseconds.
+# The ``formatter.offsetformats`` is how the "offset" string on the right
+# of the axis is formatted.  This is usually much more verbose than the tick
+# labels. Finally, the ``formatter.zeroformats`` are the formats of the
+# ticks that are "zeros".  These are tick values that are either the first of
+# the year, month, or day of month, or the zeroth hour, minute, or second.
+# These are usually the same as the format of
+# the ticks a level above.  e.g. if the axis limts mean the ticks are mostly
+# days, then we label 1 Mar 2005 simply with a "Mar".  If the axis limits
+# are mostly hours, we label Feb 4 00:00 as simply "Feb-4".
+#
+# Note that these format lists can also be passed to `.ConciseDateFormatter`
+# as optional kwargs.
+#
+# Here we modify the labels to be day-month-year, instead of the ISO
```

Comment:
"day month year" instead of "year-month-day"?

Reply:
You just mean w/o the dashes?

Reply:
yes, sorry :)

Reply:
Done...